io.c

```diff
@@ -238,6 +238,8 @@ static int io_ring_ctx_init(struct io_ring_ctx *ctx)
 {
 	int i;
 
+	memset(ctx, 0, offsetof(struct io_ring_ctx, miscdev));
```

Comment:
can we please add a comment in io.h for the io_ring_ctx, that all fields below miscdev are persisted, and above are reset every open. Keep this comment within the struct, so anyone to update it will understand the significance of new fields that go here.

Reply:
added comment